### PR TITLE
Reject configuration with both relay metadata security schemes and gateways

### PR DESCRIPTION
**Pull request by amarthadan**
[AN-405](https://api3dao.atlassian.net/browse/AN-405)

### CODE REVIEW

File: packages/airnode-validator/src/config/config.ts [hunk 1]
```diff
@@ -319,7 +344,9 @@ export const configSchema = z
   .strict()
   .superRefine(validateSecuritySchemesReferences)
   .superRefine(validateTemplateSchemes)
-  .superRefine(validateTriggersReferences);
+  .superRefine(validateTriggersReferences)
+  // TODO: Add check that the security schemes referenced in security are present
```
Comment (amarthadan) [inline packages/airnode-validator/src/config/config.ts:348]: @Siegrift Do we track this somewhere?

**Review by Siegrift — CHANGES_REQUESTED** [review]
I'd suggest improving the naming, but otherwise LGTM.

File: packages/airnode-validator/src/config/config.ts (hunk 1 above)
Comment (amarthadan) [reply packages/airnode-validator/src/config/config.ts:348]: Not really. That one is checking that if the security scheme is specified you have API credentials specified as well. I'm talking about a check that if you have something defined in `security` it needs to be available in `components.securitySchemes`. This check will just ignore if the security scheme defined in `security` doesn't exist.

Comment (Siegrift) [reply packages/airnode-validator/src/config/config.ts:348]: I see. This is checked in https://github.com/api3dao/airnode/blob/reject-relay-metadata-gateway/packages/airnode-validator/src/ois/ois.ts#L178

Comment (amarthadan) [reply packages/airnode-validator/src/config/config.ts:348]: Oh, ok, didn't notice that. In that case, I will have to change the code a bit. I thought that if it's caught early on I don't need to check for the field existence. But apparently the check is there and I still need to :smile:

**Review by Siegrift — APPROVED** [review]
👍 LGTM

**Pull request merged**